### PR TITLE
Update multiple images at once

### LINKED ISSUE #10: Explore the ability to update multiple images at once
Currently, Shipper fetches the file, updates the given image, and then submits it back to the VCS service. This means that updating multiple images in the same file requires 2 HTTP requests per image.

We should investigate if there is a practical/ergonomic way to provide multiple images to Shipper to be updated in the same batch.

### PR DESCRIPTION
**Pull request by Hamcha**
### Description

Makes most command line flags be able to be specified multiple times to allow for bulk updates with a single commit

eg.

```bash
shipper -p helm --helm-values-file helm/values.yml \
  --helm-image-path image.repository --helm-tag-path image.tag --container-image test --container-tag mytag \
  --helm-image-path image2.repository --helm-tag-path image2.tag --container-image othertest --container-tag othertag \
  --repo-kind gitea --repo-branch master --commit-author "Test"
```

You can mix & match which to specify once and which to specify many times, but every tag *must* be specified either 1 or N times, with some exceptions where some paired tags must always be both specified the same amount of times, here's some examples:

  - ✔️ one `--helm-values-file` but many `--helm-image-path` (all changes will be applied to the same YAML file)
  - ✔️ many `--helm-values-file` and `--helm-image-path` (every change will be applied to a specific file, the same file can be specified multiple times)
  - ❌ 3 instances of `--helm-image-path` but 2 instances of `--helm-values-file`
  - ❌ non-same amount of `--container-image`, `--container-tag`, `--helm-image-path`, `--helm-tag-path`

### References

Closes #10 

### Testing

- [x] This change updates test to run with new internal APIs
- [x] This change adds new tests for new edge cases

### Checklist

- [x] I have added documentation for new/changed functionality in this or a different PR.
- [x] I have signed off my commits as required by the [DCO](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin)
- [x] The correct base branch is being used, if not `main`

### CODE REVIEW

File: cmd/shipper/main.go [hunk 1]
```diff
@@ -166,14 +195,14 @@ func main() {
 				EnvVars: []string{"SHIPPER_COMMIT_MESSAGE"},
 				Value:   "Deploy",
 			},
-			&cli.StringFlag{
+			&cli.StringSliceFlag{
 				Name:     "container-image",
 				Aliases:  []string{"ci"},
 				Usage:    "Container image",
 				EnvVars:  []string{"SHIPPER_CONTAINER_IMAGE"},
```
Comment (MatteoJoliveau) [inline cmd/shipper/main.go:202]:
```suggestion
				EnvVars:  []string{"SHIPPER_CONTAINER_IMAGES"},
```

I assume this becomes a comma-separated list? e.g. `SHIPPER_CONTAINER_IMAGES=image-1,image-2`

File: templater/kustomize/kustomize.go [hunk 2]
```diff
@@ -8,79 +8,95 @@ import (
...
+		if !ok {
+			images = []interface{}{}
+		}
 
-	// Check for existing entries
-	found := false
-	for index := range imageList {
-		current, ok := imageList[index].(map[string]interface{})
+		// Make sure existing value is an array
+		imageList, ok := images.([]interface{})
```
Comment (MatteoJoliveau) [inline templater/kustomize/kustomize.go:47]: nitpick: could we type it better? I'm always scared by raw `interface{}` types as they can hide bugs

**Comment by MatteoJoliveau** [comment]
question: how does updating multiple files at once work? Does it do a single commit with all the changes or one commit per file?

File: templater/kustomize/kustomize.go (hunk 2 above)
Comment (Hamcha) [reply templater/kustomize/kustomize.go:47]: We can't, it will only match `[]interface{}` (or `[]any`, should probably change to that) if it's an array, this is how YAML/Reflection work, not my choice :C

**Comment by Hamcha** [comment]
> question: how does updating multiple files at once work? Does it do a single commit with all the changes or one commit per file?

The command always only does a single commit, regardless of how many files are in it

File: templater/kustomize/kustomize.go (hunk 2 above)
Comment (MatteoJoliveau) [reply templater/kustomize/kustomize.go:47]: Argh, fine then

**Review by MatteoJoliveau — APPROVED** [review]
LGTM 🚀
